sidecar/internal/util/util_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+)
+
+func TestValidateControllerEndpoint(t *testing.T) {
+	type args struct {
+		endpoint string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "IPv4 valid case",
+			args: args{
+				endpoint: "192.168.61.228:8080",
+			},
+			wantErr: false,
+		},
+		{
+			name: "IPv6 valid case",
+			args: args{
+				endpoint: "2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b:8080",
```

Comment:
Also include the notation that is is like `[2001:0db8::1]:8080`, see https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers

Reply:
> Also include the notation that is is like `[2001:0db8::1]:8080`, see https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers

This will fail, but i expect PodIP to be of the form `"2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b"`, so we should be fine

Reply:
If the port is included, the IPv6 address should be within `[...]` ? I do not think there is a notation without `[...]` that can include the port.

Reply:
Maybe it is required to have separate arguments for IP-address and port?

Or the endpoint needs to be a full URI, like `tcp://<IP>:<PORT>`, where `<IP>` is a valid IPv4 or IPv6 address? That way standard `net` utilities can parse it.

But with the full URI syntax, it may be difficult to create the URI in a deployment/daemonset?